sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1639,16 +1654,70 @@ impl<'ir> AsmBuilder<'ir> {
             }
         }
     }
+}
+
+fn ir_constant_to_ast_literal(constant: &Constant) -> Literal {
+    match &constant.value {
+        ConstantValue::Undef => unreachable!("Cannot convert 'undef' to a literal."),
+        ConstantValue::Unit => Literal::U64(0), // No unit.
+        ConstantValue::Bool(b) => Literal::Boolean(*b),
+        ConstantValue::Uint(n) => Literal::U64(*n),
+        ConstantValue::B256(bs) => Literal::B256(*bs),
+        ConstantValue::String(str) => Literal::String(crate::span::Span {
+            span: pest::Span::new(std::sync::Arc::from(str.as_str()), 0, str.len()).unwrap(),
+            path: None,
+        }),
+        ConstantValue::Array(_) => unimplemented!(),
```

Comment:
Can we remove `unimplemented!()` and instead give back an `Internal` compiler error message?

Reply:
I started to get this to return an error and gave up after a bit.  The diff to get an error from this deep would probably be several times larger than this PR is now.

The thing is, this is semi-temporary code.  While we have the hybrid old-ASMgen-with-optional-IR-ASMgen we're still sharing certain bits of code, including the 'data section' stuff.  The only reason we need to convert an IR type to an AST literal is to put it in the data section which is then used in ASMgen to line up consts at the end of the code and store a pointer to them.

Once we switch to IR by default and remove the old ASMgen, this won't be needed any more, there will be no need for this function.  I have a note about it in #738.